core/src/main/java/hudson/FilePath.java

```diff
@@ -748,17 +793,25 @@ public OutputStream compress(OutputStream out) throws IOException {
     public void untarFrom(InputStream _in, final TarCompression compression) throws IOException, InterruptedException {
         try {
             final InputStream in = new RemoteInputStream(_in, Flag.GREEDY);
-            act(new SecureFileCallable<Void>() {
-                public Void invoke(File dir, VirtualChannel channel) throws IOException {
-                    readFromTar("input stream",dir, compression.extract(in));
-                    return null;
-                }
-                private static final long serialVersionUID = 1L;
-            });
+            act(new UntarFrom(compression, in));
         } finally {
             _in.close();
         }
     }
+    private class UntarFrom extends SecureFileCallable<Void> {
+        private final TarCompression compression;
+        private final InputStream in;
+        UntarFrom(TarCompression compression, InputStream in) {
```

Comment:
Must be `RemoteInputStream` IMHO

Reply:
This is a `private` class. The caller passes a working stream.